test/integration/es/esClient.js

```diff
@@ -0,0 +1,114 @@
+      const location = { longitude: -1.568446, latitude: 53.805765 };
+      const fri13at2359 = moment('2017-10-15 23:59:30');
+      esClient.getOpenPharmacies(fri13at2359, location).then((pharmacies) => {
+        expect(pharmacies).to.be.an('array');
+        expect(pharmacies.length).to.be.greaterThan(0);
+        expect(pharmacies[0].name).to.equal('Hyde Park Pharmacy');
+        done();
+      }).catch(done);
+    });
+    it('should return nearest open pharmacy when none are closed', (done) => {
+      const location = { longitude: -0.973733449344135, latitude: 53.9657775613804 };
+      const friday2pm = moment('2017-09-15 14:00:00');
+      esClient.getOpenPharmacies(friday2pm, location).then((pharmacies) => {
+        expect(pharmacies).to.be.an('array');
+        expect(pharmacies.length).to.be.greaterThan(0);
+        expect(pharmacies[0].name).to.equal('Dunnington Pharmacy');
+        done();
+      }).catch(done);
+    });
+
+    it('should return pharmacies with an alteration that is open', (done) => {
+      const location = { longitude: -1.3601, latitude: 50.9778 };
+      const alterationsOpen = moment('2017-04-15 11:00:00');
+      esClient.getOpenPharmacies(alterationsOpen, location, 25, 2).then((pharmacies) => {
+        expect(pharmacies).to.be.an('array');
+        expect(pharmacies.length).to.equal(2);
+        expect(pharmacies[0].name).to.equal('Boyatt Pharmacy');
+        expect(pharmacies[1].name).to.equal('LloydsPharmacy Inside Sainsbury\'s');
+        done();
+      }).catch(done);
+    });
+
+    it('should return pharmacies with an alteration that is closed', (done) => {
+      const location = { longitude: -1.3601, latitude: 50.9778 };
+      const alterationsClosed = moment('2017-04-15 08:45:00');
+      esClient.getOpenPharmacies(alterationsClosed, location, 25, 1).then((pharmacies) => {
+        expect(pharmacies).to.be.an('array');
+        expect(pharmacies.length).to.equal(1);
+        expect(pharmacies[0].name).to.equal('Boots');
+        done();
+      }).catch(done);
+    });
+  });
+
+  describe('getPharmacies', () => {
+    const location = { longitude: -1.46519099452929, latitude: 54.0095586395326 };
+    it('should return pharmacies as an array', (done) => {
+      esClient.getPharmacies(location).then((pharmacies) => {
+        expect(pharmacies).to.be.an('array');
+        expect(pharmacies.length).to.be.greaterThan(0);
+        expect(pharmacies[0].name).to.exist;
+        done();
+      }).catch(done);
+    });
+
+    it('should return pharmacies populated with distance from location', (done) => {
+      esClient.getPharmacies(location).then((pharmacies) => {
+        pharmacies.forEach(p => expect(p.distanceInMiles).to.be.greaterThan(0, `Pharmacy ${p.identifier} does not have a distance`));
+        done();
+      }).catch(done);
+    });
+
+    it('reducing the radius should return fewer pharmacies', (done) => {
```

Comment:
This test isn't reducing the radius, it is reducing the size.

Reply:
a good argument for moving towards parameter objects rather than having several parameters to a function

Reply:
@st3v3nhunt - I don't think that is correct (`getPharmacies(location, radius, size)`. Why do you think it is size not radius?

Reply:
'cuz when I wrote that, the test was executing `esClient.getPharmacies(location, resultsLimit, X)` 😜